### ISSUE TITLE
Unable to scrape socket

### ISSUE THREAD

**Issue by Misosooup** [kumina/postfix_exporter#53]
Hi,

I tried using this exporter and it started fine as a daemon. However, when it tried to scrape the showq scoket, it returns a permission denied error.

`/var/spool/postfix/public` is owned by `postfix:root` and `showq` is owned by `postfix:postfix`.

I have tried running the daemon as `postfix` user and that worked. I tried giving `postfix-exporter` sudoers access but that still returns the permission error to access showq. 

![image](https://user-images.githubusercontent.com/1595688/77617178-34778800-6f7f-11ea-9658-1ed3a2fae32b.png)

**Comment by flyhard** [comment]
What user are you running as? does that user have privileges? The user running postfix_exporter needs to have access.

**Comment by Misosooup** [comment]
Yeah, for some reason the permission was set really weirdly so instead of
creating a new user for postfix_exporter, we have decided to use the
postfix user to run postfix_exporter.

**Comment by mpavlovicbb** [comment]
I am also having the same problem.
Postfix_exporter service is created with root user and permisions on socker are:
srw-rw-rw- 1 postfix postfix 0 Mar 31  2015 /var/spool/postfix/public/showq

**Comment by learning813** [comment]
Did anyone find the reason behind this ?

**Comment by ndelic0** [comment]
@learning813 Have you tried with adding user running postfix exporter to postfix and postdrop groups?
```
usermod -a -G postfix postfix-exporter
usermod -a -G postdrop postfix-exporter
```

**Comment by learning813** [comment]
@ndelic0 : its the postfix user itself which is running postfix-exporter

**Comment by ndelic0** [comment]
check the ownership and permissions set on following dirs
```
/var/spool/postifx
/var/spool/postifx/public
```

**Comment by learning813** [comment]
@ndelic0 : `/var/spool/postifx` 

`root@postfix-01•> ls -ltr /var/spool/ | grep postfix
lrwxrwxrwx  1 root root    9 Jan 30 03:09 postfix -> /mailsdata
root@postfix-01•> ls -ltr /var/spool/postfix/ | grep public
drwx--x--- 2 postfix postdrop  88 Jan 30 03:09 public
`
postfix directory is pointing to "/mailsdata" which is owned by root:root which is same as `/var/spool/postfix` 
and inside it the folder `public` is owned by `postfix:postrdop`

**Comment by tdewolff** [comment]
I recommend to only add the exporter's user to the `postdrop` group and not the `postfix` group as well. It would open up a lot of access. E.g. I run the exporter as the `prometheus` user and do:

```bash
usermod -a -G postdrop prometheus
```

On a CentOS 9 the default ownership and permissions of `/var/spool/postfix/public` are:
```
drwx--x---.  2 postfix postdrop   88 Jun 14 23:47 .
srw-rw-rw-.  1 postfix postfix     0 Jun 14 23:47 showq
```

So you need the `postdrop` group to enter into `/var/spool/postfix/public` only, there are no other limiting permissions.

Regarding @Misosooup , not sure what is wrong there. What user does postfix run as? Why isn't `/var/spool/postfix/public` owned by postfix:postdrop? Are you sure you're running as the root account (it looks like you're not)?

**Comment by Nimon77** [comment]
Hi, I have the same problem when using docker-mailserver, did you find a fix for that ?
